librepo/handle.c

```diff
@@ -788,6 +788,29 @@ lr_handle_prepare_urls(LrHandle *handle, GError **err)
     return TRUE;
 }
 
+static gboolean
+lr_download_url_retry(LrHandle *lr_handle, const char *url, int fd,
```

Comment:
I'd prefer to preserve all arguments from the original function, ideally in the same order:
```
lr_yum_download_url(LrHandle *lr_handle, const char *url, int fd,
                    gboolean no_cache, gboolean is_zchunk, GError **err);
```
The new function could look like:
```
lr_yum_download_url_retry(int attempts, LrHandle *lr_handle, const char *url, int fd,
                    gboolean no_cache, gboolean is_zchunk, GError **err);
```

Reply:
do you want to have the `_yum_` string in function name?

Reply:
Yes, keep it there for consistency with the old function name.